openmc/deplete/operator.py

```diff
@@ -609,12 +618,11 @@ def _generate_materials_xml(self):
         through direct memory writing.
 
         """
-        materials = openmc.Materials(self.geometry.get_all_materials()
-                                     .values())
+        materials = openmc.Materials(self.materials)
 
         # Sort nuclides according to order in AtomNumber object
         nuclides = list(self.number.nuclides)
-        for mat in materials:
+        for mat in self.materials:
```

Comment:
Is `self.materials` the same as what `materials` is/was?

Reply:
I believe so, yes. `self.materials` will be gathered from `self.geometry` at a different point in time in `Operator.__init__` (if needed).

No need for the additional local `materials` variable though. Removing that...